pulsar-client-cpp/lib/ClientImpl.cc

```diff
@@ -342,10 +342,11 @@ void ClientImpl::subscribeAsync(const std::string& topic, const std::string& con
             return;
         } else if (conf.getConsumerType() == ConsumerShared) {
             ConsumersList consumers(consumers_);
-            for (ConsumersList::iterator it = consumers.begin(); it != consumers.end(); ++it) {
-                ConsumerImplBasePtr consumer = it->lock();
+            for (auto& weakPtr : consumers) {
+                ConsumerImplBasePtr consumer = weakPtr.lock();
                 if (consumer && consumer->getSubscriptionName() == consumerName && !consumer->isClosed()) {
                     lock.unlock();
+                    LOG_INFO("Reusing existing consumer instance for " << topic << " -- " << consumerName);
```

Comment:
Maybe a `warn` would be better to alert the user of a miss use?

Reply:
I would suggest to go the other way if anything. This isn't an something the user needs to be concerned about, so maybe it should be debug.

Reply:
The reason I added the log is that when looking at the logs one can now see `Subscribing.. ` but there's no corresponding `Subscribed... ` message. I think info level should be ok to keep the consistency with the rest of log messages.

Reply:
Great :+1: